test/sharness/t0114-gateway-subdomains.sh

```diff
@@ -403,6 +403,14 @@ test_hostname_gateway_response_should_contain \
   "http://127.0.0.1:$GWAY_PORT/ipns/en.wikipedia-on-ipfs.org/wiki" \
   "Location: http://en.wikipedia-on-ipfs.org.ipns.example.com/wiki"
 
+# DNSLink on Public gateway with a single-level wildcard TLS cert
+# "Option C" from  https://github.com/ipfs/in-web-browsers/issues/169
+test_expect_success \
+  "request for example.com/ipns/{fqdn} with X-Forwarded-Proto redirects to TLS-safe label in subdomain" "
+  curl -H \"Host: example.com\" -H \"X-Forwarded-Proto: https\" -sD - \"http://127.0.0.1:$GWAY_PORT/ipns/en.wikipedia-on-ipfs.org/wiki\" > response &&
+  test_should_contain \"Location: https://en-wikipedia--on--ipfs-org.ipns.example.com/wiki\" response
```

Comment:
PR in action:
1. User opens `ipns://en.wikipedia-on-ipfs.org/wiki`
2. User agent resolves URI via public gateway at  `example.com/ipns/en.wikipedia-on-ipfs.org/wiki` which returns HTTP 301 redirect to  `https://en-wikipedia--on--ipfs-org.ipns.example.com/wiki` if `X-Forwarded-Proto: https` is present.